thirtyinch/src/test/java/net/grandcentrix/thirtyinch/internal/PresenterScopeTest.java

```diff
@@ -0,0 +1,166 @@
+        };
+        scope.save("b", presenter2);
+
+        assertThat(scope.size()).isEqualTo(2);
+
+        assertThat(scope.getAll())
+                .hasSize(2)
+                .contains(presenter1, presenter2);
+    }
+
+    @Test
+    public void getAllIsEmptyNotNullForEmptyScope() throws Exception {
+        final PresenterScope scope = new PresenterScope();
+        assertThat(scope.size()).isEqualTo(0);
+        assertThat(scope.getAll()).isEqualTo(new ArrayList<>());
+    }
+
+    @Test
+    public void getAllMappingsIsEmptyNotNullForEmptyScope() throws Exception {
+        final PresenterScope scope = new PresenterScope();
+        assertThat(scope.size()).isEqualTo(0);
+        assertThat(scope.getAllMappings()).isEqualTo(new ArrayList<>());
+    }
+
+    @Test
+    public void overrideMappingThrows() throws Exception {
+
+        final PresenterScope scope = new PresenterScope();
+        final TiPresenter presenter1 = new TiPresenter() {
+        };
+        scope.save("myId", presenter1);
+
+        // override with same id throws
+        final TiPresenter presenter2 = new TiPresenter() {
+        };
+        try {
+            scope.save("myId", presenter2);
+            fail("did not throw");
+        } catch (IllegalStateException e) {
+            assertThat(e).hasMessageContaining("myId");
+        }
+    }
+
+    @Test
+    public void removeOneMapping() throws Exception {
+
+        final PresenterScope scope = new PresenterScope();
+        final TiPresenter presenter = new TiPresenter() {
+        };
+        scope.save("a", presenter);
+
+        assertThat(scope.getAll())
+                .hasSize(1)
+                .contains(presenter);
+
+        final TiPresenter removedPresenter = scope.remove("a");
+
+        assertThat(scope.getAll()).isEmpty();
+        assertThat(removedPresenter).isEqualTo(presenter);
+    }
+
+    @Test
+    public void removePresenterNotInScope() throws Exception {
+        final PresenterScope scope = new PresenterScope();
+        final TiPresenter removedPresenter = scope.remove("a");
+        assertThat(removedPresenter).isNull();
+    }
+
+    @Test
+    public void saveNullPresenterThrows() throws Exception {
+        final PresenterScope scope = new PresenterScope();
+        try {
+            scope.save("a", null);
+            fail("did not throw");
+        } catch (IllegalStateException e) {
+            assertThat(e).hasMessageContaining("null").hasMessageContaining("presenter");
+        }
+    }
+
+    @Test
+    public void savePresenterWithNullIdThrows() throws Exception {
+        final PresenterScope scope = new PresenterScope();
+        try {
+            scope.save(null, new TiPresenter() {
+            });
+            fail("did not throw");
+        } catch (IllegalStateException e) {
+            assertThat(e).hasMessageContaining("null").hasMessageContaining("id");
+        }
+    }
+
+    @Test
+    public void saveSamePresenterTwiceThrows() throws Exception {
```

Comment:
Duplicated test. This is the same like `overrideMappingThrows`, right? 🤔

Reply:
That's the reason why `Adding the presenter twice with different id` was missing. The implementation was wrong and therefore duplicated.
Fixed it :)